### ISSUE TITLE
[BUG] builtin.file pattern matches directory names, causing false positives

### ISSUE THREAD

**Issue by fabianvf** [konveyor/analyzer-lsp#1092]
### Is there an existing issue for this?

- [x] I have searched the existing issues

Related: #1083 (same root cause - patterns matching against full paths instead of just filenames)

### Konveyor version

Latest main branch

### Priority

Major

### Current Behavior

The `builtin.file` provider matches patterns against **both** the full file path AND the filename (basename). This causes false positives when directory names match part of the pattern.

**Example:**
- Rule pattern: `spring.*\.jar`
- File: `/path/to/spring-framework-petclinic/.mvn/wrapper/maven-wrapper.jar`
- Expected: No match (`maven-wrapper.jar` doesn't start with "spring")
- Actual: **Matches!** because "spring" appears in directory name `spring-framework-petclinic`

### Expected Behavior

Patterns like `spring.*\.jar` should only match files whose **name** starts with "spring", not files located in directories containing "spring".

### How Reproducible

Always

### Steps To Reproduce

**1. Create directory structure:**
```
test-app/
└── spring-framework-petclinic/     # "spring" in directory name
    └── .mvn/wrapper/
        └── maven-wrapper.jar       # Should NOT match spring.*\.jar
```

**2. Create test rule (`rules/test-rule.yaml`):**
```yaml
- ruleID: path-bug-spring-match
  description: "BUG: This incorrectly matches maven-wrapper.jar"
  labels:
  - konveyor.io/include=always
  when:
    builtin.file:
      pattern: spring.*\.jar
```

**3. Create provider settings (`provider-settings.json`):**
```json
[
  {
    "name": "builtin",
    "initConfig": [
      {
        "location": "./test-app/spring-framework-petclinic"
      }
    ]
  }
]
```

**4. Run analyzer:**
```bash
./analyzer \
    --provider-settings provider-settings.json \
    --rules rules/test-rule.yaml \
    --output-file output.yaml \
    --label-selector "konveyor.io/include=always"
```

**5. Actual result:**
```yaml
- name: konveyor-analysis
  tags:
  - Spring
  insights:
    path-bug-spring-match:
      description: 'BUG: This incorrectly matches maven-wrapper.jar'
      incidents:
      - uri: file:///tmp/test-app/spring-framework-petclinic/.mvn/wrapper/maven-wrapper.jar
        message: ""
```

**Expected result:** No match / rule should be in `unmatched` list.

### Root Cause

In `provider/lib.go`, the `filterFilesByPathsOrPatterns` function (line 235):

```go
regex.MatchString(file) || regex.MatchString(filepath.Base(file))
```

The pattern is matched against the **full path first**. The regex `spring.*\.jar` matches:
1. `spring` in path component `spring-framework-petclinic`
2. `.*` matches rest of path including `/framework-petclinic/.mvn/wrapper/maven-wrapper`
3. `\.jar` matches at the end

### Suggested Fix

**Option 1:** For `builtin.file`, match against `filepath.Base(file)` **only** (simplest fix)

**Option 2:** Add `filenameOnly: true` option to `builtin.file` condition for backward compatibility

**Option 3:** Document the behavior and require rules to use anchored patterns like `(?:^|[/\\])spring[^/\\]*\.jar$`

### Affected Rules in konveyor/rulesets

- `spring-catchall-00001`: `spring.*\.jar`
- `embedded-cache-libraries-*`: `.*<name>.*\.jar$` patterns
- Many other file pattern rules

### Environment

macOS, tested with analyzer built from main branch

### Anything else?

This was discovered while testing Hub integration. Different directory structures between Hub and CLI cause different false positives on the same application (spring-framework-petclinic), making it difficult to create stable test expectations.

**Comment by eemcmullan** [comment]
Resolved in https://github.com/konveyor/analyzer-lsp/pull/1106
